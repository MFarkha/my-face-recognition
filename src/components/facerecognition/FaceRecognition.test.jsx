import React from 'react';
import {render, screen} from '@testing-library/react';

import FaceRecognition from './FaceRecognition';

it('renders and displays an image correctly', () => {
    const mockImageUrl = '';
    const mockBoxes = [];
    const { container } = render (<FaceRecognition boxes={mockBoxes} imageUrl={mockImageUrl} />);
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(screen.queryAllByRole('bounding-box')).toHaveLength(0);
    expect(container).toMatchSnapshot();
})

it('renders and displays a bounding box of image correctly', () => {
    const mockImageUrl = 'http://some.image';
    const mockBoxes = [{
        topRow: 0,
        bottomRow: 100,
        rightCol: 200,
        leftCol: 0
    }];
    const { container } = render (<FaceRecognition boxes={mockBoxes} imageUrl={mockImageUrl} />);
    expect(container).toMatchSnapshot();

})
