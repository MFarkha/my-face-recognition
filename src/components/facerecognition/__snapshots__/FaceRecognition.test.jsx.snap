// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and displays a bounding box of image correctly 1`] = `
<div>
  <div
    class="center ma"
  >
    <div
      class="absolute mt2"
    >
      <img
        alt=""
        height="auto"
        id="inputImage"
        src="http://some.image"
        width="500px"
      />
      <div
        class="bounding-box"
        id="0"
        style="top: 0px; right: 200px; bottom: 100px; left: 0px;"
      />
    </div>
  </div>
</div>
`;

exports[`renders and displays an image correctly 1`] = `
<div>
  <div
    class="center ma"
  >
    <div
      class="absolute mt2"
    >
      <img
        alt=""
        height="auto"
        id="inputImage"
        src=""
        width="500px"
      />
    </div>
  </div>
</div>
`;
